programs/KJVDataDump/main.cpp

```diff
@@ -371,14 +396,44 @@ int main(int argc, char *argv[])
 						if (bOutputTemplates) {
 							std::cout << pVerse->m_strTemplate.toUtf8().data() << std::endl;
 						}
-						if (bOutputVerseText) {
-							if (bOutputTemplates) std::cout << strSpacer.toUtf8().data();
-							if (bOutputTransChangeAdded) {
-								std::cout << lstVerseWords.join(QChar(' ')).toUtf8().data();
-							} else {
-								std::cout << strParsedVerse.toUtf8().data();
+						if (bOutputVerseText && nLetterCount > 0) {
+							QString verse = strParsedVerse.toUtf8().data();
+							QString clean = QString(verse).toLower().simplified().replace(QRegularExpression("[\\[\\]()\.,:;!?]"), "");
+							QString cleanNoSpaces = QString(clean).remove(' ');
+							int basicVowelCount = clean.count(QRegularExpression("[aeiou]"));
+							int yCount = clean.count(QRegularExpression("y$|y "));
+							int consonantCount = cleanNoSpaces.length() - (basicVowelCount + yCount);
+
+							bool qPrint = true;
+
+							if (cleanNoSpaces.length() == nLetterCount) {
+
+								QStringList list = clean.split(QRegularExpression("\\s"));
+
+								foreach(QString item, list) {
```

Comment:
When iterating over constant data like this, it would be better to use a const reference to avoid unnecessary copy operations:

```suggestion
								foreach(const QString &item, list) {
```